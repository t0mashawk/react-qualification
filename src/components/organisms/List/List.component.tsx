import React from 'react'
import * as Styled from './List.styles'
import ListItem from '../../molecules/ListItem/ListItem.component'

interface ListProps{
    items:any
}

function List(props:ListProps){
    const {items} = props
    return(
        <Styled.Container>
             {items
            
            .map((item: any, index:number) => (
                <ListItem href={item.html_url} text={item.name} buttonText="like" key={index}/>
            ))}
        </Styled.Container>
    )
}

export default List