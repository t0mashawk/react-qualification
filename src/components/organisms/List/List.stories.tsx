import React from "react";
import { storiesOf } from "@storybook/react";
import List from "./List.component";

const listItems = [
    { name: "test1", html_url: "test1" },
    { name: "test2", html_url: "test2" },
    { name: "test11", html_url: "test11" },
  ]

storiesOf("List", module)
  .add("default", () => (
    <List
      items={listItems}
    />
  ))

