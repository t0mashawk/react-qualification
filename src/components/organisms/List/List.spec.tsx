import React from "react";
import renderer from "react-test-renderer";

import List from "./List.component";

test("List renders as expected", () => {
    const listItems = [
        { name: "test1", html_url: "test1" },
        { name: "test2", html_url: "test2" },
        { name: "test11", html_url: "test11" },
      ]

  const component = renderer.create(<List items={listItems} />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});