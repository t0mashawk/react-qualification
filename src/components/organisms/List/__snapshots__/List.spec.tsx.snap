// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders as expected 1`] = `
<ul
  className="sc-AxjAm eFaybh"
>
  <li
    className="sc-AxiKw jdKcko"
  >
    <a
      href="test1"
    >
      test1
    </a>
    <div
      className="sc-AxirZ jsCLat"
    >
      <button
        onClick={[Function]}
      >
        like
      </button>
    </div>
  </li>
  <li
    className="sc-AxiKw jdKcko"
  >
    <a
      href="test2"
    >
      test2
    </a>
    <div
      className="sc-AxirZ jsCLat"
    >
      <button
        onClick={[Function]}
      >
        like
      </button>
    </div>
  </li>
  <li
    className="sc-AxiKw jdKcko"
  >
    <a
      href="test11"
    >
      test11
    </a>
    <div
      className="sc-AxirZ jsCLat"
    >
      <button
        onClick={[Function]}
      >
        like
      </button>
    </div>
  </li>
</ul>
`;
